### ISSUE TITLE
mnist loss

### ISSUE THREAD

**Issue by kikyou123** [dbbert/dfn#4]
elif options['loss'] == 'binary_crossentropy':
            # Clipping to avoid NaN's in binary crossentropy: https://github.com/Lasagne/Lasagne/issues/436
            output_frame = T.clip(output_frame, np.finfo(np.float32).eps, 1-np.finfo(np.float32).eps)
            frame_loss = binary_crossentropy(output_frame, target[:,[i],:,:])

if you use binary_crossentropy , should you promise the target is 0 or 1 ? but the target is float in [0,1].
and i think the output should using sigmoid function to promise [0,1]

**Comment by stephenjia** [comment]
We did not use any activation function here because we model a pixel in a future frame as a linear combination of neighboring pixels in the previous frame. If we added sigmoid, then it would a little different from our original idea.
